prism/src/main/java/org/apache/falcon/FalconServer.java

```diff
@@ -88,6 +94,14 @@ public static void main(String[] args) throws Exception {
             appPath = cmd.getOptionValue(APP_PATH);
         }
 
+        String isSafeMode = "false";
+        if (cmd.hasOption(SAFE_MODE)) {
+            isSafeMode = cmd.getOptionValue(SAFE_MODE);
+            validateSafemode(isSafeMode);
+        }
+        LOG.info("Setting system property Safemode to : {}", isSafeMode);
+        System.setProperty(StartupProperties.SAFEMODE_PROPERTY, isSafeMode);
```

Comment:
@bvellanki    I am wondering whether we need a bit stronger mechanism for keeping track of  safemode internally.   For example, if someone adds the java system property to the Java command line, we will be in safe mode.   Wondering whether we can track it by tracking it in a system file - something that is persistent also and enables safemode to survive restarts as a potential enhancement